dev_guide/app_tutorials/ruby_on_rails.adoc

```diff
@@ -0,0 +1,357 @@
+
+* POSTGRESQL_DATABASE
+* POSTGRESQL_USER
+* POSTGRESQL_PASSWORD
+
+By setting these variables when creating the PostgreSQL database, the image will ensure that:
+
+* A database exists with the specified name
+* A user exists with the specified name
+* The user can access the specified database with the specified password
+
+To do that type following command with your desired values:
+
+----
+$ oc new-app postgresql -e POSTGRESQL_DATABASE=db_name -e POSTGRESQL_USER=username -e POSTGRESQL_PASSWORD=password
+----
+
+In case you would like to also set password for the database administrator, append the previous command with:
+----
+-e POSTGRESQL_ADMIN_PASSWORD=admin_pw
+----
+
+To watch the progress of this command type:
+
+----
+$ oc get pods --watch
+----
+
+=== Creating frontend service
+
+To bring your application to OpenShift, you need to specify a repository in which your application lives, using once again the link:../new_app.html[oc new-app] command, in which you will need to specify database related link:../../db_images/postgresql.html#environment-variables[environment variables] we setup in the link:./ruby_on_rails.html#creating-database-service[creating database service] section:
+
+----
+$ oc new-app path/to/source/code --name=rails-app -e POSTGRESQL_USER=username -e POSTGRESQL_PASSWORD=password -e POSTGRESQL_DATABASE=db_name
+----
+
+With this command OpenShift will fetch the source code, setup the Builder image, link:../builds.html[build] your application image and also will deploy the newly created image together with the specified link:../../db_images/postgresql.html#environment-variables[environment variables]. Application is going to be named `rails-app`.
+
+You can verify the environment variables have been added by viewing the JSON document of the `rails-app` link:../deployments.html[DeploymentConfig]:
+
+----
+$ oc get dc rails-app -o json
+----
+
+You should see the following section:
+
+----
+env": [
+    {
+        "name": "POSTGRESQL_USER",
+        "value": "username"
+    },
+    {
+        "name": "POSTGRESQL_PASSWORD",
+        "value": "password"
+    },
+    {
+        "name": "POSTGRESQL_DATABASE",
+        "value": "db_name"
+    }
+],
+----
+
+To check the link:../builds.html[build] process use the link:../builds.html#accessing-build-logs[build-logs] command:
+
+----
+$ oc build-logs rails-app-1
+----
+
+Once the link:../builds.html[build] is complete, you can look at the running link:../../architecture/core_concepts/pods_and_services.html#pods[pods] in OpenShift
+
+----
+$ oc get pods
+----
+
+You should see a line starting with myapp-(#number)-(some hash) and that is your application running in OpenShift.
+
+To migrate your database you either need to do it:
```

Comment:
change this to:  Before your application will be functional, you need to initialize the database by running the database migration script.  There are two ways you can do this: